File: ios/App/AppUITests/ManagerModeTests.swift
import XCTest

/*
  Automated QA Foundation Batch(2026-09-18) — 요청 2-C: 회원 ↔ 관리자 모드 전환.

  관리자 모드 진입은 active manager_centers 소속(=센터 운영자 계정)이 있어야만
  가능하다(ACL-005, app/mypage/page.tsx의 profile?.isManager 가드). 기존 테스트
  계정 중 TEST_MANAGER_A가 이 조건을 만족한다는 전제로, 이메일 로그인 후 확인한다
  (외부 OAuth는 사용 안 함 — 요청 4번 원칙). 테스트 계정 env가 없으면 스킵한다.
*/
final class ManagerModeTests: XCTestCase {
    override func setUpWithError() throws {
        continueAfterFailure = false
    }

    func testSwitchToManagerModeAndBack() throws {
        guard TestAccount.hasManagerA else {
            throw XCTSkip("TEST_MANAGER_A_EMAIL/PASSWORD가 설정되지 않아 건너뜀 — docs/AUTOMATED_QA.md 참고")
        }
        let app = XCUIApplication()
        app.launch()
        loginWithEmail(app, email: TestAccount.managerAEmail!, password: TestAccount.managerAPassword!)

        // 로그인 성공 시 handleLogin()이 window.location.href = "/"로 이동 — 홈 화면이
        // 다시 뜨는 걸로 로그인 완료를 판단한다.
        assertExists(app.webViews.staticTexts["오늘은 어떤 움직임을 찾나요?"], "로그인 후 홈 화면 복귀")

        let myTab = app.webViews.links["마이"]
        assertExists(myTab, "하단 nav '마이' 탭")
        myTab.tap()

        // app/mypage/page.tsx — "관리자 모드로 전환"(manager-mode-switch)
        let switchToManager = app.webViews.links["관리자 모드로 전환"]
        assertExists(switchToManager, "'관리자 모드로 전환' 링크(관리자 권한 있는 계정에서만 노출)")
        switchToManager.tap()

        // app/manager/page.tsx — "관리자 모드" 라벨 + "회원 모드로 전환 ↩" 복귀 링크
        assertExists(app.webViews.staticTexts["관리자 모드"], "관리자 홈 상단 '관리자 모드' 라벨")
        let switchBack = app.webViews.links["회원 모드로 전환 ↩"]
        assertExists(switchBack, "'회원 모드로 전환' 복귀 링크")
        switchBack.tap()

        assertExists(app.webViews.staticTexts["오늘은 어떤 움직임을 찾나요?"], "회원 모드 복귀 후 홈 화면")
    }
}

File: android/app/src/androidTest/java/com/mwhabit/app/ManagerModeTests.java
package com.mwhabit.app;

import static androidx.test.espresso.web.sugar.Web.onWebView;
import static androidx.test.espresso.web.webdriver.DriverAtoms.findElement;
import static androidx.test.espresso.web.webdriver.DriverAtoms.webClick;
import static org.junit.Assert.assertTrue;
import static org.junit.Assume.assumeTrue;

import androidx.test.espresso.web.webdriver.Locator;
import androidx.test.ext.junit.rules.ActivityScenarioRule;
import androidx.test.ext.junit.runners.AndroidJUnit4;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

/**
 * Automated QA Foundation Batch(2026-09-18) — 요청 2-C: 회원 ↔ 관리자 모드 전환.
 *
 * TEST_MANAGER_A_EMAIL/PASSWORD가 없으면 건너뛴다(assumeTrue — JUnit에서 "실패"가 아닌
 * "skipped"로 리포트됨, iOS의 XCTSkip과 동일한 효과).
 */
@RunWith(AndroidJUnit4.class)
public class ManagerModeTests {

    @Rule
    public ActivityScenarioRule<MainActivity> activityRule = new ActivityScenarioRule<>(MainActivity.class);

    @Test
    public void switchToManagerModeAndBack() {
        assumeTrue(
            "TEST_MANAGER_A_EMAIL/PASSWORD가 설정되지 않아 건너뜀 — docs/AUTOMATED_QA.md 참고",
            TestSupport.TestAccount.hasManagerA()
        );
        TestSupport.waitForInitialLoad();
        TestSupport.loginWithEmail(TestSupport.TestAccount.managerAEmail(), TestSupport.TestAccount.managerAPassword());

        assertTrue("로그인 후 홈 화면 복귀 실패", TestSupport.webElementExists(".header .location"));

        onWebView()
            .withElement(findElement(Locator.CSS_SELECTOR, "a.nav-item[href='/mypage']"))
            .perform(webClick());
        try { Thread.sleep(2000); } catch (InterruptedException ignored) { Thread.currentThread().interrupt(); }

        // app/mypage/page.tsx — <a class="list-row manager-mode-switch" href="/manager">
        assertTrue(
            "'관리자 모드로 전환' 링크(.manager-mode-switch)를 찾지 못함 — 계정에 관리자 권한이 없을 수 있음",
            TestSupport.webElementExists("a.manager-mode-switch")
        );
        onWebView()
            .withElement(findElement(Locator.CSS_SELECTOR, "a.manager-mode-switch"))
            .perform(webClick());
        try { Thread.sleep(2000); } catch (InterruptedException ignored) { Thread.currentThread().interrupt(); }

        // app/manager/page.tsx — <div class="mgr-mode-bar">, 복귀 링크 <a class="mgr-mode-switch">
        assertTrue("관리자 홈 상단 바(.mgr-mode-bar)를 찾지 못함", TestSupport.webElementExists(".mgr-mode-bar"));
        assertTrue("'회원 모드로 전환' 링크(.mgr-mode-switch)를 찾지 못함", TestSupport.webElementExists("a.mgr-mode-switch"));

        onWebView()
            .withElement(findElement(Locator.CSS_SELECTOR, "a.mgr-mode-switch"))
            .perform(webClick());
        try { Thread.sleep(2000); } catch (InterruptedException ignored) { Thread.currentThread().interrupt(); }

        assertTrue("회원 모드 복귀 후 홈 화면 확인 실패", TestSupport.webElementExists(".header .location"));
    }
}
